src/System.Management.Automation/engine/LanguagePrimitives.cs

```diff
@@ -5550,6 +5557,29 @@ private static bool TypeConverterPossiblyExists(Type type)
             return false;
         }
 
+        private static bool TypeConverterPossiblyExists(Type fromType, Type toType)
```

Comment:
I move the check in original method but tests failed ("Multiple ambiguous overloads found for "SetBufferContents"")

Reply:
Should I revert?

Reply:
It's very possible that adding `TypeDescriptor.GetConverter` in `TypeConverterPossibleExists` causes a behavior change of this method for some types -- the method used to return `false` for those types and now return `true`.
And it might be because of the intrinsic types there because none of them has `TypeConverterAttribute`.

https://github.com/PowerShell/PowerShell/blob/70ab772da77d8cc1cbec2e3aa29bd328ff1de589/src/System.Management.Automation/engine/LanguagePrimitives.cs#L5726-L5728

This concerns me, because it means the assumption of this if check will be changed once we moved to `TypeDescriptor.GetConverter`.
It would be good to understand what the root cause of the `Multiple ambiguous overloads found for "SetBufferContents"` error.

Reply:
> It would be good to understand what the root cause of the Multiple ambiguous overloads found for "SetBufferContents" error.

In Clear-Host we call the method and first argument is a Hashtable. We find a conversion from the Hashtable to Host.Coordinates type or Host.BufferCell or Host.BufferCell[,] type. It is ConvertViaNoArgumentConstructor. But then TypeConverterPossiblyExists() return true for Hashtable and we replace already found converter with ConvertCheckingForCustomConverter where converter is CollectionConverter. I guess the converter can convert Hashtable to Host.BufferCell and Host.BufferCell[,] types and we get ambiguous overloads error.